### PR TITLE
Update config and reference tests to v1.0.0-rc.0 + Reject BLS infinity pubkeys and signatures

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

This is the continuation of @ajsutton PR #2952: 
- Reject BLS infinity pubkeys and signatures
- Allow zero private key creation but throw when signing with this key 
- Add `BLSConstants.VALID_INFINITY` temporary compatibility flag which may be set to `true` to preserve previous BLS behavior
- Make reference BLS tests covering both BLS implementations 
- Mikuli implementation code clean up: many methods removed and/or inlined 
- Adjust unit tests for the new BLS infinity treatment

## Fixed Issue(s)
Part of #2951 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

**Comment by Nashatyrev** [comment]
We just need to decide when and where to adjust the `BLSConstants.VALID_INFINITY`

**Comment by ajsutton** [comment]
> We just need to decide when and where to adjust the `BLSConstants.VALID_INFINITY`

I was thinking we'd put it in `Constants` and then it could be set via the network definition file.  Would be a non-standard option in network config, default to rejecting infinity as per 1.0 spec but our `medalla.yaml` could set it to allow infinity just on Medalla (and probably also Zinken but it will die soon).

**Comment by Nashatyrev** [comment]
> I was thinking we'd put it in Constants

@ajsutton do you think that https://github.com/PegaSysEng/teku/pull/2971/commits/c80cdb9c73d9f99d7f82b1ee5d717c14b21bb45e crutch is ok as a temporary solution?

**Comment by ajsutton** [comment]
> > I was thinking we'd put it in Constants
> 
> @ajsutton do you think that [c80cdb9](https://github.com/PegaSysEng/teku/commit/c80cdb9c73d9f99d7f82b1ee5d717c14b21bb45e) crutch is ok as a temporary solution?

Yep, that's exactly what I had in mind.  Bit ugly but effective. :)

File: util/src/main/java/tech/pegasys/teku/util/config/Constants.java [hunk 1]
```diff
@@ -167,13 +169,19 @@
   public static final int VALID_VALIDATOR_SET_SIZE = 10000;
   public static final int NETWORKING_FAILURE_REPEAT_INTERVAL = 3; // in sec
 
+  // Custom
+  // Temporary BLS backward compatibility flag
+  // Set to 1 if required BLS to be pre rc-1 spec compatible
+  public static int BLS_INFINITY_VALID = 0;
```
Comment (ajsutton) [inline util/src/main/java/tech/pegasys/teku/util/config/Constants.java]: Can we make this a boolean instead of 0 or 1?

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**